tests/common.sh

```diff
@@ -134,7 +134,7 @@ function build_docs()
 function run_host_tests()
 {
     pushd host
-    make
+    make ARCH32=0 CI
```

Comment:
Is there any reason not to only do testing in 32-bit mode?  Seems like it's as close as we're likely to get, and should remove the need for all of the casting dances you had to do with `(uint32_t)(unsigned long)&abcd` throughout the core.

Reply:
We can make sure the host CI tests have 32-bit dev libs if needed.  We're already installing valgrind and lcov, if that was your concern.

Reply:
This is the reason. CI was using native compiler, it still does. It can be later sorted out.
```
valgrind:  Fatal error at startup: a function redirection
valgrind:  which is mandatory for this platform-tool combination
valgrind:  cannot be set up.  Details of the redirection are:
valgrind:  
valgrind:  A must-be-redirected function
valgrind:  whose name matches the pattern:      strlen
valgrind:  in an object with soname matching:   ld-linux.so.2
valgrind:  was not found whilst processing
valgrind:  symbols from the object with soname: ld-linux.so.2
valgrind:  
valgrind:  Possible fixes: (1, short term): install glibc's debuginfo
valgrind:  package on this machine.  (2, longer term): ask the packagers
valgrind:  for your Linux distribution to please in future ship a non-
valgrind:  stripped ld.so (or whatever the dynamic linker .so is called)
valgrind:  that exports the above-named function using the standard
valgrind:  calling conventions for this platform.  The package you need
valgrind:  to install for fix (1) is called
```